src/main/java/cinema/security/CustomUserDetailsService.java

```diff
@@ -0,0 +1,32 @@
+package cinema.security;
+
+import static org.springframework.security.core.userdetails.User.withUsername;
+
+import cinema.model.User;
+import cinema.service.UserService;
+import java.util.NoSuchElementException;
+import org.springframework.security.core.userdetails.User.UserBuilder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.stereotype.Service;
+
+@Service
+public class CustomUserDetailsService implements UserDetailsService {
+    private final UserService userService;
+
+    public CustomUserDetailsService(UserService userService) {
+        this.userService = userService;
+    }
+
+    @Override
+    public UserDetails loadUserByUsername(String email) {
+        User user = userService.findByEmail(email).orElseThrow(()
+                -> new NoSuchElementException("Can`t find user by email " + email));
```

Comment:
let's use here spring exception UsernameNotFoundException (it's default behaviour and will be more obvious)

Reply:
Fixed